hoomd/hpmc/integrate_nec.py

```diff
@@ -0,0 +1,300 @@
+
+    def _process_chain_time(self, value):
+        if 0.0 <= value:
+            return value
+        else:
+            raise ValueError(
+                "chain_time has to be positive (got {}).".format(value))
+
+    def _process_update_fraction(self, value):
+        if 0.0 < value <= 1.0:
+            return value
+        else:
+            raise ValueError(
+                "update_fraction has to be between 0 and 1. (got {})".format(
+                    value))
+
+    def _attach(self):
+        super()._attach()
+
+    @property
+    def nec_counters(self):
+        """Trial move counters.
+
+        The counter object has the following attributes:
+
+        * ``chain_start_count``:        `int` Number of chains
+        * ``chain_at_collision_count``: `int` Number of collisions
+        * ``chain_no_collision_count``: `int` Number of chain events that are
+          no collision (i.e. no collision partner found or end of chain)
+        * ``distance_queries``:         `int` Number of sweep distances
+          calculated
+        * ``overlap_errors``:           `int` Number of errors during sweep
+          calculations
+
+        Note:
+            The counts are reset to 0 at the start of each
+            `hoomd.Simulation.run`.
+        """
+        if self._attached:
+            return self._cpp_obj.getNECCounters(1)
+        else:
+            return None
+
+    @log
+    def virial_pressure(self):
+        """float: virial pressure.
+
+        Note:
+            The statistics are reset at every timestep.
+        """
+        if self._attached:
+            return self._cpp_obj.virial_pressure
+        else:
+            return None
+
+    @log
+    def particles_per_chain(self):
+        """float: particles per chain.
+
+        Note:
+            The statistics are reset at every `hoomd.Simulation.run`.
+        """
+        if self._attached:
+            necCounts = self._cpp_obj.getNECCounters(1)
+            return (necCounts.chain_at_collision_count * 1.0
+                    / necCounts.chain_start_count)
+        else:
+            return None
+
+    @log
+    def chains_in_space(self):
+        """float: rate of chain events that did neither collide nor end.
+
+        Note:
+            The statistics are reset at every `hoomd.Simulation.run`.
+        """
+        if self._attached:
+            necCounts = self._cpp_obj.getNECCounters(1)
+            return (necCounts.chain_no_collision_count
+                    - necCounts.chain_start_count) / (
+                        necCounts.chain_at_collision_count
+                        + necCounts.chain_no_collision_count)
+        else:
+            return None
+
+
+class Sphere(HPMCNECIntegrator):
+    """HPMC chain integration for spheres (2D/3D).
+
+    Args:
+        d (float): Maximum move displacement, Scalar to set for all types,
+            or a dict containing {type:size} to set by type.
+        chain_time (float): length of a chain in units of time.
```

Comment:
See new unit documentation formatting (examples can be found in `hoomd/hpmc/integrate.py`.